designdocs/provider_contexts.md

```diff
@@ -0,0 +1,379 @@
+            default_name: application_name,
+            localized_names: BTreeMap::new(), // Initially empty, can be updated later
+            default_logo_250_100_png_bytes: BoundedVec::default(), // Initially empty, can be updated later
+            localized_logo_250_100_png_bytes: BTreeMap::new(), // Initially empty, can be updated later
+        };
+
+        ProviderToApplicationRegistryEntry::<T>::insert(provider_id, application_identifier, application_entry);
+    ```
+
+5. `propose_to_add_application` will also insert the logo hashes into the `ApprovedLogoHashes` storage map.
+6. Introduce a new extrinsic to `update_logo` for updating `ApprovedLogos` map (post goveranance registration) provided with logo bytes.
+
+    ```rust
+        #[pallet::call_index(14)]
+        pub fn update_logo(
+            origin: OriginFor<T>,
+            logo_bytes: BoundedVec<u8, T::MaxProviderLogo250X100Size>,
+        ) -> DispatchResultWithPostInfo {
+            // Store the logo bytes in ApprovedLogos
+        }
+    ```
+
+    Note: This extrinsics will compute the logo hash and update the `ApprovedLogos` storage map if the hash exists in `ApprovedLogoHashes`.
+7. Ensure that application updates would require governance approval, hence `propose_to_add_application` will be used for both adding and updating applications.
+
+### **Storage Migration** <a id='migration'></a>
+
+To support the new structure, a storage migration will be required to:
+
+- Migrate existing `ProviderToRegistryEntry` to the new format with default values for localized names and logos.
+
+### **Runtime RPC Changes** <a id='rpc'></a>
+
+1. Introduce a new RPC method "get_provider_application_context" to fetch the application context for a given provider and application index.
+
+    ```rust
+    // Return a constructed ApplicationContext struct
+    // This struct will contain the default name, localized names, default logo, and localized logos.
+    #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+    pub struct ApplicationContext<T:Config> {
+        pub default_name: BoundedVec<u8, T::MaxProviderNameSize>,
+        pub localized_names: BTreeMap<BoundedVec<u8, T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderNameSize>>,
+        pub default_logo_250_100_png: BoundedVec<u8, T::MaxProviderLogo250X100Size>,
+        pub localized_logo_250_100_png: BTreeMap<BoundedVec<u8, T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderLogo250X100Size>>,
+    }
+
+    // The rpc should fetch the data from `ProviderToApplicationRegistryEntry` and construct the ApplicationContext with logos from `ApprovedLogos`.
+    #[rpc]
+    pub fn get_provider_application_context(
+        provider_id: ProviderId,
+        application_index: ApplicationIndex,
+    ) -> Option<ApplicationContext<T>>;
+    ```
+
+2. Optionally, introduce "get_provider" RPC method to include the new fields for localized names and logos and return a provider registry entry.
```

Comment:
I like the idea of one that pulls the info for the `application` and `locale` and one that just pulls all the data.